### PR TITLE
kie-issues#1245: Adapt KIE Tools release jobs to sign and upload release candidate artifacts to https://dist.apache.org/repos/dist/dev/incubator/kie

### PR DESCRIPTION
**Pull request by rodrigonull**
This PR introduces a new Jenkins job to build, sign and publish release candidate artifacts to: https://dist.apache.org/repos/dist/dev/incubator/kie as part of the issue: [#1245](https://github.com/apache/incubator-kie-issues/issues/1245)

### CODE REVIEW

**Review by porcelli — APPROVED** [review]

File: .ci/jenkins/Jenkinsfile.release-candidate [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+        stage('Update kogito version') {
+            steps {
+                dir('kie-tools') {
+                    script {
+                        buildUtils.pnpmUpdateKogitoVersion(params.RELEASE_CANDIDATE_VERSION, params.RELEASE_CANDIDATE_VERSION)
+                    }
+                }
+            }
+        }
```
Comment (tiagobento) [inline .ci/jenkins/Jenkinsfile.release-candidate:132]: Also do `pnpm update-stream-name-to <stream-name>`

E.g., 10.0.x or main or 10.0.0
Used to replace “latest” or “daily-dev” or "10.0.x"

Comment (tiagobento) [reply .ci/jenkins/Jenkinsfile.release-candidate:132]: Please do that on the setup-branch job as well.

File: .ci/jenkins/Jenkinsfile.release-candidate [hunk 2]
```diff
@@ -114,7 +115,17 @@ pipeline {
...
+
+        stage('Update stream name') {
+            steps {
+                dir('kie-tools') {
+                    script {
+                        buildUtils.pnpmUpdateStreamName(params.BRANCH_NAME)
```
Comment (tiagobento) [inline .ci/jenkins/Jenkinsfile.release-candidate]: params.RELEASE_VERSION

File: .ci/jenkins/release-jobs/Jenkinsfile.chrome-extensions [hunk 3]
```diff
@@ -350,8 +351,8 @@ pipeline {
             }
             steps {
                 script {
-                    env.CHROME_EXTENSION_RELEASE_ZIP_FILE = "incubator-kie-${params.RELEASE_VERSION}-business-automation-chrome-extension.zip"
-                    env.SWF_CHROME_EXTENSION_RELEASE_ZIP_FILE = "incubator-kie-${params.RELEASE_VERSION}-sonataflow-chrome-extension.zip"
+                    env.CHROME_EXTENSION_RELEASE_ZIP_FILE = "incubator-kie-${params.RELEASE_CANDIDATE_VERSION}-business-automation-chrome-extension.zip"
+                    env.SWF_CHROME_EXTENSION_RELEASE_ZIP_FILE = "incubator-kie-${params.RELEASE_CANDIDATE_VERSION}-sonataflow-chrome-extension.zip"
```
Comment (tiagobento) [inline .ci/jenkins/release-jobs/Jenkinsfile.chrome-extensions:355]: params.RELEASE_CANDIDATE_VERSION is `10.0.0` or `10.0.0-rc1`?

**Review by tiagobento — CHANGES_REQUESTED** [review]

File: .ci/jenkins/release-jobs/Jenkinsfile.chrome-extensions (hunk 3 above)
Comment (rodrigonull) [reply .ci/jenkins/release-jobs/Jenkinsfile.chrome-extensions:355]: `10.0.0-rc1`

**Review by tiagobento — APPROVED** [review]

**Review by porcelli — APPROVED** [review]

**Review by baldimir — APPROVED** [review]

**Pull request merged**